src/master-playlist-controller.js

```diff
@@ -1064,26 +1064,34 @@ export class MasterPlaylistController extends videojs.EventTarget {
     }
 
     let isFinalRendition = this.masterPlaylistLoader_.isFinalRendition_();
+    let playlists = this.masterPlaylistLoader_.master.playlists;
 
-    if (isFinalRendition) {
-      // Never blacklisting this playlist because it's final rendition
+    if (playlists.length === 1) {
+      // Never blacklisting this playlist because it's the only playlist
       videojs.log.warn('Problem encountered with the current ' +
-                       'HLS playlist. Trying again since it is the final playlist.');
+                       'HLS playlist. Trying again since it is the only playlist.');
 
       this.tech_.trigger('retryplaylist');
       return this.masterPlaylistLoader_.load(isFinalRendition);
     }
+    if (isFinalRendition) {
+      playlists.forEach((playlist) => {
+        // clear the blacklist duration for the other playlists when
```

Comment:
Rather than describing what the code is doing, it might make more sense to have a comment about the reason we are doing it. Something like `Since we're on the final non-blacklisted playlist, and we're about to blacklist it, instead of erring the player or retrying this playlist, we clear out the current blacklist so the other playlists may be attempted in case any have been fixed.`